src/plugins/BlockingCommandPlugin.ts

```diff
@@ -0,0 +1,187 @@
+import { AbstractPlugin, IMetaData } from '../plugin/AbstractPlugin'
+import PluginAPI from '../plugin/PluginAPI'
+import { IJID } from '../JID.interface';
+import Translation from '@util/Translation';
+import { IContact } from '@src/Contact.interface';
+import { IMessage } from '@src/Message.interface';
+import { Status } from '../vendor/Strophe'
+import showContactBlockDialog from '../ui/dialogs/contactBlock'
+import Roster from '../ui/Roster'
+
+const MIN_VERSION = '4.0.0';
+const MAX_VERSION = '99.0.0';
+
+const NAMESPACE_BLOCKING_COMMAND = 'urn:xmpp:blocking'
+
+export default class BlockingCommandPlugin extends AbstractPlugin {
+
+   private blocklist : string[];
+   private static instance: BlockingCommandPlugin;
+
+   public static getId(): string {
+      return 'blocking-command';
+   }
+
+   public static getName(): string {
+      return 'Blocking Command';
+   }
+
+   public static getMetaData(): IMetaData {
+      return {
+         description: Translation.t('setting-vcard-avatar-enable'),
+         xeps: [{
+            id: 'XEP-0191',
+            name: 'Blocking Command',
+            version: '1.3',
+         }]
+      }
+   }
+
+   public static getInstance()
+   {
+       return BlockingCommandPlugin.instance;
+   }
+
+   constructor(pluginAPI: PluginAPI) {
+      super(MIN_VERSION, MAX_VERSION, pluginAPI);
+
+      let connection = pluginAPI.getConnection();
+
+      BlockingCommandPlugin.instance=this;
+
+      pluginAPI.addFeature(NAMESPACE_BLOCKING_COMMAND);
+
+      connection.registerHandler(this.onReceiveBlockingItems, NAMESPACE_BLOCKING_COMMAND, 'iq');
+
+      let _self = this;
+      pluginAPI.registerConnectionHook((status, condition) => {
+         if (status === Status.ATTACHED) {
+             // result in onReceiveBlockingItems
+            _self.getBlocklist();
+         }
+      });
+
+      pluginAPI.addAfterReceiveMessageProcessor(this.onMessageIncomingProcessor);
+
+      let roster = Roster.get();
```

Comment:
This could maybe instantiate the roster to early. Not sure if this is used somewhere else in the code.

Reply:
Any suggestions?

Reply:
No, I have to check that by my own. It was just a remark.

Reply:
moved addMenuEntry for BlockDialog into connectionHook, so it will be…
d5686c2

… added on event CONNECTED (only once)